torch/fx/graph_module.py

```diff
@@ -296,18 +301,65 @@ def recompile(self) -> None:
         cls = type(self)
         cls.forward = _forward_from_src(self._code)
 
-        cls_call = cls.__call__
+        # If we're calling `recompile` on a GraphModule that has already
+        # been recompiled, `cls.__call__` would be `wrapped_call`.
+        # We don't want to wrap `wrapped_call` again (as this would
+        # ultimately lead to O(num_recompiles) "unwrappings" later), so
+        # we use the instance attribute `_cls_call` to store the
+        # original implementation of `__call__`
+        if not self._cls_call:
+            self._cls_call = cls.__call__
 
         def print_full_traceback(exctype, value, tb):
             traceback.print_exception(exctype, value, tb)
 
+        # Previously, if an error occurred when valid
+        # symbolically-traced code was run with an invalid input, the
+        # user would see the source of the error as coming from
+        # `File "<eval_with_key_N">`, where N is the number of times
+        # `recompile` had been called on the GraphModule. We use this
```

Comment:
`N` is actually the result of a global counter, so only monotonic with how many times `recompile` was called on this specific GraphModule but not 1:1